File: frontend/src/types/resume.ts
export interface Resume {
  id: string;
  user_id: string;
  file_name: string;
  parsed_text?: string | null;
  created_at: string;
}

export interface ResumeListResponse {
  resumes: Resume[];
}

export interface ResumeUploadResponse {
  resume: Resume;
}

File: api-gateway/internal/model/resume.go
package model

import "time"

type Resume struct {
	ID         string    `json:"id"`
	UserID     string    `json:"user_id"`
	FileName   string    `json:"file_name"`
	FilePath   string    `json:"-"`
	ParsedText *string   `json:"parsed_text,omitempty"`
	CreatedAt  time.Time `json:"created_at"`
}
